### PR TITLE
TracerProvider shutdown release resources

### LINKED ISSUE #3534: Document TracerProvider Shutdown as releasing resources
- [ ] Update [this](https://github.com/open-telemetry/opentelemetry-go/blob/4e763472eee5ef03b97274694d2d1b07f0b3cd11/sdk/trace/provider.go#L230) to:
  > // Shutdown shuts down TracerProvider. All registered span processors are shut down in the order they were registered and any held computational resources are released.
- [ ] Unallocate the span processors from the `TracerProvider` after shutdown to release any held resource.
  - Ensure this works with all other `TracerProvider` methods

_Originally posted by @MrAlias in https://github.com/open-telemetry/opentelemetry-go/pull/3524#discussion_r1044858763_

### PR DESCRIPTION
**Pull request by fatsheep9146**
fix #3534

Signed-off-by: Ziqi Zhao <zhaoziqi9146@gmail.com>

### CODE REVIEW

**Review by hanyuancheung — APPROVED** [review]
lgtm

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Add `Producer` interface and `Reader.RegisterProducer(Producer)` to `go.opentelemetry.io/otel/sdk/metric` to enable external metric Producers. (#3524)
 - Add `NewMetricProducer` to `go.opentelemetry.io/otel/bridge/opencensus`, which can be used to pass OpenCensus metrics to an OpenTelemetry Reader. (#3541)
 - Global logger uses an atomic value instead of a mutex. (#3545)
+- Document TracerProvider Shutdown releasing resources. (#3551)
```
Comment (dashpole) [inline CHANGELOG.md:27]: This is more than just documentation
```suggestion
- TracerProvider Shutdown releases references to spanProcessors. (#3551)
```

File: sdk/trace/provider.go [hunk 2]
```diff
@@ -255,6 +256,7 @@ func (p *TracerProvider) Shutdown(ctx context.Context) error {
 			}
 		}
 	}
+	p.spanProcessors.Store(spanProcessorStates{})
```
Comment (dashpole) [inline sdk/trace/provider.go:259]: Do we also need to ensure new span processors cannot be registered?

Comment (fatsheep9146) [reply sdk/trace/provider.go:270]: Shutdown is able to prevent future registered span processors

**Review by dashpole — APPROVED** [review]

**Comment by fatsheep9146** [comment]
@MrAlias could you help review this pr?

File: CHANGELOG.md [hunk 3]
```diff
@@ -24,6 +24,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Add `Producer` interface and `Reader.RegisterProducer(Producer)` to `go.opentelemetry.io/otel/sdk/metric` to enable external metric Producers. (#3524)
 - Add `NewMetricProducer` to `go.opentelemetry.io/otel/bridge/opencensus`, which can be used to pass OpenCensus metrics to an OpenTelemetry Reader. (#3541)
 - Global logger uses an atomic value instead of a mutex. (#3545)
+- TracerProvider Shutdown releases references to spanProcessors. (#3551)
```
Comment (MrAlias) [inline CHANGELOG.md:27]:
```suggestion
- The `Shutdown` method of the `"go.opentelemetry.io/otel/sdk/trace".TracerProvider` releases all computational resources when called the first time. (#3551)
```

**Review by MrAlias — APPROVED** [review]

**Pull request merged**